frontend/src/components/FileUpload.tsx

```
@@ -0,0 +1,201 @@
+      {isUploading && (
+        <div className="upload-progress">
+          <div className="progress-bar">
+            <div className="progress-fill" style={{ width: `${uploadProgress}%` }} />
+          </div>
+          <span className="progress-text">Uploading... {uploadProgress}%</span>
+        </div>
+      )}
+
+      {response && (
+        <div className="upload-success" role="status">
+          <span className="success-icon">✓</span>
```

Biome (2.1.2)

[error] The elements with this role can be changed to the following elements:
<output>

For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)